test/unit/plugins/rest_spec.js

```diff
@@ -95,6 +104,25 @@ describe("The Rest plugin", function () {
 		});
 
 		describe("with valid parameters passed", function () {
+
+			before(function () {
+				startSshPollingStub = Sinon.stub(sshAdapter, "startSshPolling")
+				.returns(Bluebird.resolve());
+
+				updateInstanceStub = Sinon.stub(instanceAdapter,"updateInstance")
+				.returns(Bluebird.resolve("asdasd"));
+
+				getPublicIPAddressStub = Sinon.stub(awsAdapter, "getPublicIPAddress", function () {
+					return (Bluebird.resolve(VALID_IP_ADDRESS));
+				});
+			});
+
+			after(function () {
+				getPublicIPAddressStub.restore();
+				startSshPollingStub.restore();
+				updateInstanceStub.restore();
```

Comment:
Sharing stubs like this has a tendency to create false positives. It is very easy to forget to call `.restore()` and to not know that you are working with state from a previous test case. It would be better if we could find a way to rework these test cases so that they don't share state.